blueprints/addon/index.js

```diff
@@ -49,6 +49,9 @@ module.exports = {
     contents.dependencies = contents.dependencies || {};
     contents.devDependencies = contents.devDependencies || {};
 
+		// addon should depend on the ember-source that matches ember-cli
+    contents.dependencies['ember-source'] = contents.devDependencies['ember-cli'];
```

Comment:
```suggestion
    contents.peerDependencies['ember-source'] = `>= ${contents.devDependencies['ember-source']}`;
```

I think this is what you want. This means that the consuming app chooses what ember-source, and the addon is expected to be compatible with at least this version